plugins/mac-videotoolbox/encoder.c

```diff
@@ -75,16 +90,35 @@ static void log_osstatus(int log_level, struct vt_encoder *enc,
 	CFRelease(err);
 }
 
-static CFStringRef obs_to_vt_profile(const char *profile)
+static CFStringRef obs_to_vt_profile(CMVideoCodecType codec_type,
+				     const char *profile)
 {
-	if (strcmp(profile, "baseline") == 0)
+	if (codec_type == kCMVideoCodecType_H264) {
+		if (strcmp(profile, "baseline") == 0)
```

Comment:
@gxalpha Mark for refactoring later: Unify the pattern used here (we have `if() return` as well as `if() return; else if() return(); [..]` patterns) and also default fallthrough returns.

Will not hold up merging, but should be cleaned up later.